weather/utils.py

```diff
@@ -0,0 +1,41 @@
+from datetime import datetime
+import os
+from typing import Callable
+import json
+
+from fastapi import HTTPException
+
+
+def check_write_access():
+    filepath = os.getcwd()
+
+    if os.access(filepath, os.W_OK):
+        print("Access granted")
+    else:
+        print("Denied")
```

Comment:
The `check_write_access` function prints 'Access granted' or 'Denied' based on write permissions. Consider using logging instead of print statements for better control over output, especially in production environments.